### PR TITLE
IsAllDataRequested true after ending

### PR DESCRIPTION
**Pull request by cijothomas**
Opening this to get feedback on this spec requirement.
Given the fact that .NET's Activity doesn't become read-only after ending, this is not really having an impact.

### CODE REVIEW

File: src/OpenTelemetry/Trace/TracerProviderSdk.cs [hunk 1]
```diff
@@ -97,6 +97,10 @@ internal class TracerProviderSdk : TracerProvider
                         return;
                     }
 
+                    // Spec says IsRecording must be true once span ends.
+                    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#isrecording
```
Comment (reyang) [inline src/OpenTelemetry/Trace/TracerProviderSdk.cs:101]: I guess the spec says it should be false?

Comment (cijothomas) [reply src/OpenTelemetry/Trace/TracerProviderSdk.cs:101]: Oh my bad! (time to stop working on Friday :D )

Comment (reyang) [reply src/OpenTelemetry/Trace/TracerProviderSdk.cs:101]: I would vote for leaving this block as commented out:
1. `IsAllDataRequested` has a slightly different semantic than `IsRecording`.
2. We don't see a use case, except for burning extra CPU cycle.

Comment (CodeBlanch) [reply src/OpenTelemetry/Trace/TracerProviderSdk.cs:101]: Agree with @reyang, seems a bit silly to do this 😜

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**